### DISCUSSION TITLE
Custom gamemodes on Android

### DISCUSSION THREAD

**Discussion by Neon095** [ppy/osu#16503] [Q&A] [answered]
I tried to add custom gamemodes to android and i have seen it is impossible. I tried to add the custom modes extracting the apk to the assemblies files but it doesn't work. Is there any form to actually do it?
[runtime.log](https://github.com/ppy/osu/files/7888834/runtime.log)
[performance.log](https://github.com/ppy/osu/files/7888836/performance.log)
[database.log](https://github.com/ppy/osu/files/7888837/database.log)
[network.log](https://github.com/ppy/osu/files/7888843/network.log)

**Comment by frenzibyte** [comment]
To clarify, you already tried adding the rulesets to the `rulesets` folder in the osu!lazer directory (`/Android/data/sh.ppy.osulazer/files/rulesets`), right? I don't see any attempts of loading a custom ruleset.

**Reply by Neon095** [reply]
No im gonna try that rn, i didnt know that folder exist

**Reply by Neon095** [reply]
Ok i tried setakki and it didn't worked showed an error in notifications, then i tried adding swing too and didn't worked too, also the game crashed with it
Here the logs:
[database.log](https://github.com/ppy/osu/files/7888921/database.log)
[network.log](https://github.com/ppy/osu/files/7888923/network.log)
[performance.log](https://github.com/ppy/osu/files/7888924/performance.log)
[runtime.log](https://github.com/ppy/osu/files/7888925/runtime.log)

**Reply by frenzibyte** [reply]
This log looks incomplete to me? Not even a sign of intro playing. In general it may be better to wait until rulesets are distributed properly as mentioned below.

**Comment by peppy** [comment, marked as answer]
Generally we can't support custom rulesets on mobile until we are distributing them ourselves.

**Reply by Neon095** [reply]
But if someone do a mod especially for android it will work?
